### PR TITLE
[FLOC-3117] Use a wait instead of polling for container stop.

### PR DESCRIPTION
**Pull request by itamarst**

### CODE REVIEW

File: flocker/node/_docker.py [hunk 1]
```diff
@@ -710,8 +693,7 @@ def _remove():
                 # attempting removal or use -f")'
                 # This code should probably be removed once the above
                 # issue has been resolved. See [FLOC-1850]
-                while self._blocking_container_runs(container_name):
-                    sleep(0.01)
+                self._client.wait(container_name)
```
Comment (adamtheturtle) [inline flocker/node/_docker.py]: :question: The description of this mentions a 404 error, but the issue comments describe that it is a 500 error now.

**Comment by adamtheturtle** [comment]
Thanks @itamarst , ideally this would come with some benchmarking numbers, or an explanation of why this would use less CPU. I understand that we are in a rush now so merge when you are confident that this is better for CPU usage and Buildbot is happy.

**Comment by exarkun** [comment]
Still seems to work: I set up an add/remove loop and didn't encounter any cases where Docker was complaining about this new remove logic.  Code makes sense, too, and seems like the most straightforward implementation anyway (if you know about `wait`, at least).  Will merge when buildbot shows me I haven't overlooked something else.

**Comment by exarkun** [comment]
Lots of problems on buildbot.  None clearly attributable to this change.  Merging.

**Pull request merged**